### DISCUSSION TITLE
p8 file Invalid Api Key

### DISCUSSION THREAD

**Discussion by rakeshadvani1000** [codemagic-ci-cd/codemagic-docs#2682] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

1. In Team Settings > Integrations, click "Connect" for Developer Portal
2. I added Issuer ID, Key ID, Name and uploaded the P8 key file itself.
3. Every time, I upload it says "Invalid API Key". I checked the file and it has no spaces and it is direct download from Apple connect.

### Expected results

P8 file should get uploaded and integration should get completed.

### Actual results

Getting error - "Invalid API Key".

**Comment by dtrdic** [comment]
hey @rakeshadvani1000 I suggest regenerating the API key in App Store Connect and ensuring that the role of the key is set to App Manager or Admin. Also, make sure you are using Team keys, not Individual keys.
![Screenshot 2024-08-20 at 09 05 29](https://github.com/user-attachments/assets/537bc22c-c7c3-4ebb-9dbb-86c8c99d921a)

**Comment by rakeshadvani1000** [comment]
Hi @dtrdic - Thank you for your reply. 

I generated the key again and made sure to use Team Keys and App Manager role. However, that did not work either. 

Just a note that the key is being created by my client on a different machine (Mac) and I am uploading it from Linux (he sent the key via email). Let me know if any specific permissions are required for the key. 

Thank you.

**Reply by dtrdic** [reply]
hey @rakeshadvani1000 you need to make sure that the `issuerID`(e.g _b46b8317-3a60-4ab6-af64-6c3f046398d2_) and the `keyID`(e.g. _CTBX1234D_) are correct. And that the keyID is of that exact .p8 file you have downloaded.
![Screenshot 2024-08-30 at 10 26 43](https://github.com/user-attachments/assets/e6ace39b-545c-45d2-b7f4-a2a507b6c0bd)

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
